detekt/wrapper/src/main/java/io/buildfoundation/bazel/detekt/ExecutionUtils.java

```diff
@@ -0,0 +1,92 @@
+package io.buildfoundation.bazel.detekt;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class ExecutionUtils {
+    /**
+     * Returns true if run-as-test-target flag is included in arguments
+     */
+    public static boolean shouldRunAsTestTarget(List<String> arguments) {
+        return arguments.contains("--run-as-test-target");
+    }
+
+    /**
+     * Retrieves the output path for the test result from the input arguments.
+     */
+    public static String getExecutionResultOutputPath(List<String> inputArgs) {
+        String outputPath = getArgument(inputArgs, "--execution-result");
+        if (outputPath == null) {
+            System.err.println("File path for execution-result was not set");
+            System.exit(1);
+        }
+        return outputPath;
+    }
+
+    /**
+     * Writes the execution result to a file
+     */
+    public static void writeExecutionResultToFile(Integer exitCode, String executionResultOutputPath) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(executionResultOutputPath))) {
+            writer.write(String.format("#!/bin/bash\n\nexit %d\n", exitCode));
```

Comment:
This should probably be written in Starlark instead of from Java. You can return the exit code written to a file and then cat that in a Starlark action that just calls exit with what ever value it reads.

Reply:
Is this what you were thinking? https://github.com/davidkurkov/bazel_rules_detekt/pull/1/commits/1b53f5b273b2ba8f693c161e53d9271b91df30d5